src/os/bluestore/BlueFS.cc

```diff
@@ -1477,7 +1477,7 @@ int BlueFS::_flush_range(FileWriter *h, uint64_t offset, uint64_t length)
   h->buffer_appender.flush();
 
   bool buffered;
-  if (h->file->fnode.ino == 1)
+  if (h->file->fnode.ino == 1 || (h->writer_type == WRITER_WAL && cct->_conf->get_val<bool>("bluefs_wal_direct_io")))
```

Comment:
please replace the get_val<> with a local variable and make the config observer update it when the config changes.  get_val<> is too slow to call on every _flush_range.  see https://github.com/ceph/ceph/pull/18958